src/test/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/integration/VersionServiceIntegrationTest.kt

```diff
@@ -0,0 +1,121 @@
+class VersionServiceIntegrationTest : IntegrationTestBase() {
+
+  @Autowired
+  lateinit var planProgressNoteRepository: PlanProgressNotesRepository
+
+  @Autowired
+  lateinit var versionService: VersionService
+
+  @Autowired
+  lateinit var planRepository: PlanRepository
+
+  @Autowired
+  lateinit var planVersionRepository: PlanVersionRepository
+
+  @Autowired
+  lateinit var goalRepository: GoalRepository
+
+  // from SQL scripts
+  val testPlanUuid = UUID.fromString("556db5c8-a1eb-4064-986b-0740d6a83c33")
+  val testPlanVersionUuid = UUID.fromString("9f2aaa46-e544-4bcd-8db6-fbe7842ddb64")
+  var newPlanVersionUuid: UUID = UUID.randomUUID()
+
+  @Test
+  @Order(1)
+  @DisplayName("Create new PlanVersion")
+  @WithMockUser(username = "UserId|Username")
+  @Sql(
+    scripts = [
+      "/db/test/oasys_assessment_pk_data.sql",
+      "/db/test/goals_data.sql",
+      "/db/test/related_area_of_need_data.sql",
+      "/db/test/step_data.sql",
+      "/db/test/plan_notes_data.sql",
+    ],
+    executionPhase = BEFORE_TEST_METHOD,
+  )
+  @Sql(
+    scripts = [
+      "/db/test/plan_notes_cleanup.sql",
+      "/db/test/step_cleanup.sql",
+      "/db/test/related_area_of_need_cleanup.sql",
+      "/db/test/goals_cleanup.sql",
+      "/db/test/oasys_assessment_pk_cleanup.sql",
+    ],
+    executionPhase = AFTER_TEST_METHOD,
+  )
+  fun `test new plan version has the same associations as previous version`() {
+    // a plan and a version are added via SQL scripts in annotation
+
+    // create a new version
+    val planVersionOne = versionService.createNewPlanVersion(testPlanVersionUuid)
+    newPlanVersionUuid = planVersionOne.uuid
+
+    // now check there are two versions
+    assertThat(planVersionRepository.findAll().size).isEqualTo(2)
+
+    // check that when fetched the plan references the version with ID 1
+    val planEntity = planRepository.findByUuid(testPlanUuid)
+    assertThat(planEntity.currentVersion!!.version).isEqualTo(1)
+
+    // check that the UUID of version 0 is now different to the original UUID
+    val planVersionZero = planVersionRepository.findByPlanUuidAndVersion(testPlanUuid, 0)
+    assertThat(planVersionZero.uuid).isNotEqualTo(testPlanVersionUuid)
+
+    // check that both versions reference the same plan
+    assertThat(planVersionZero.planId).isEqualTo(planVersionOne.planId)
+
+    // check that the latest version references the plan and that the previous version is null (reference is owned by PlanEntity)
+    assertThat(planVersionOne.plan!!).isNotNull
+    assertThat(planVersionZero.plan).isNull()
+
+    // check that each version has two goals
+    assertThat(planVersionZero.goals.size).isEqualTo(planVersionOne.goals.size)
+    assertThat(goalRepository.findAll()).hasSize(4)
+
+    // check that the first step in the first goal of each plan version have matching descriptions but different UUIDs
+    val planVersionZeroFirstGoal = goalRepository.findByUuidWithSteps(planVersionZero.goals.toList().get(0).uuid)
```

Comment:
Use indexing instead of `.get`

Reply:
Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan/pull/104/commits/35c138c3305c49a6e5495fbdba9018d327b2cba8